.github/actions/validate-access/action.yaml

```diff
@@ -0,0 +1,53 @@
+name: Validate Access
+description: 'Check if the workflow is triggered by an admin user'
+
+# This callable workflow checks if the workflow is triggered by
+# a code owner or an image maintainer by testing the github.actor
+# variable against the CODEOWNERS file and the contacts.yaml file
+# under oci/* path
+
+inputs:
+  admin-only:
+    description: 'The protected workflow should only be triggered as a code owner or an image maintainer'
+    required: true
+    default: 'false'
+  image-path:
+    description: 'The path to the image to be built'
+    required: true
+  github-token:
+    description: 'The GH_TOKEN for the GitHub CLI'
+    required: true
+
+runs:
+  using: "composite" 
+  steps:
+    - name: Check if the workflow is triggered by an admin user
+      shell: bash
+      env:
+        GH_TOKEN: ${{ inputs.github-token }}
+      run: |
+        echo "github.actor: ${{ github.actor }}"
```

Comment:
I think we should break this out into a script and add some testing to ensure that it fails safe. I'm currently adding testing elsewhere with pytest, maybe we can integrate that somehow. Otherwise there is also https://github.com/sstephenson/bats

We could also start the script `set -e` for added safety.

Reply:
Breaking it into a standalone script is a good idea. Setting `-e` is not necessary, as running the composite action with `bash` is actually `bash --noprofile --norc -e -o pipefail`, which already sets this.